### ISSUE TITLE
New buzzwords This Week in Programming: Real-Time Collaborative Coding Comes to Atom and Visual Studio

### ISSUE THREAD

**Issue by postmillenial** [postmillenial/reading-list#1604]
This Week in Programming: Real-Time Collaborative Coding Comes to Atom and Visual Studio<br>
http://ift.tt/2zQozZO<br><img width="640" height="480" src="http://ift.tt/2jD5G6z" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2017/11/25ef53e9-4278905797_64ca333e46_b.jpg"><p>Most office workers these days couldn&rsquo;t imagine their modern workplace without real-time collaboration tools like Google Docs or Microsoft Office 365. For many developers, however, real-time collaboration often comes in the form of Slack, siloed IDEs and screen sharing. While there has been an increasing number of <a href="http://ift.tt/2hJPzDI" class="ext-link" rel="external nofollow">IDEs that offer real-time collaboration</a> in recent years, some of the most popular have lagged behind &mdash; until now.</p>
<p>While new versions of languages and features come out all the time&nbsp;&mdash; and this week is no exception&nbsp;&mdash; the highlight of programming news this time around just might be the arrival of real-time collaboration to Atom and Visual Studio. As the Atom blog post notes, &ldquo;the logistics of writing code with another programmer can be such a hassle that many people don&rsquo;t bother.&rdquo; This week, both GitHub and Microsoft announced (on the same day, at that!) <a href="http://ift.tt/2iXc9W3" class="ext-link" rel="external nofollow">Teletype for Atom</a> and <a href="http://ift.tt/2ihdgAi" class="ext-link" rel="external nofollow">Live Share for Visual Studio and Visual Studio Code</a>, respectively.</p>
<p>Already a popular open source code editor, Atom just earlier this year saw the <a href="http://ift.tt/2jq23Af" class="local-link">release of a full set of IDE features</a>. Now, Teletype for Atom offers &ldquo;a new way to dive right into code with remote collaborators [&hellip;] in real time with your own configurations in your own programming environment on any file you can open in Atom.&rdquo;</p>
<p>Everything is free and open source, as usual, and fully encrypted, so none of your files or edits are shared with GitHub&rsquo;s servers. Perhaps most excitingly, the entire thing is built with an editor-agnostic library called teletype-client, which should make inter-operation between different text editors possible in the future. While still in beta, Teletype for Atom is <a href="http://ift.tt/2z4atVI" class="ext-link" rel="external nofollow">available for download</a> and the <a href="http://ift.tt/2ijJ73o" class="ext-link" rel="external nofollow">Atom blog post</a> walks you through installation.</p>
<p>As for <a href="http://ift.tt/2ihdgAi" class="ext-link" rel="external nofollow">Visual Studio Live Share</a>, real-time code editing is just the beginning. Not only can your teammate &ldquo;read the code you shared without having to clone a repo or install any dependencies your code relies on,&rdquo; but &ldquo;they can use rich language features to navigate within the code; not only just opening other files as text but using semantic analysis-based navigation like Go to Definition or Peek.&rdquo; Beyond that, you can even debug code collaboratively. Unfortunately, Live Share isn&rsquo;t yet available, but you can <a href="http://ift.tt/2zMFaeZ" class="ext-link" rel="external nofollow">sign up for a sneak peek preview</a>. Until then, you can also take a look at a video of the new features in action:</p>
<p></p>
<h2>This Week in Programming News</h2>
<ul>
<li>
<strong>Kotlin/Native v0.4:</strong> Last week, Kotlin creator JetBrains announced that it would support iOS with the next version of Kotlin/Native. This week, the company announced the full release of <a href="http://ift.tt/2infa2y" class="ext-link" rel="external nofollow">Kotlin/Native v0.4</a>. Highlights of the release include support for accessing Objective-C APIs on iOS and macOS, experimental support of <a href="http://ift.tt/1fXcXqh" class="ext-link" rel="external nofollow">WebAssembly</a>, and &ldquo;major changes making app development in Kotlin/Native way easier.&rdquo; For everything that happened with Kotlin last week at its conference, refer to last week&rsquo;s edition of this column, titled &ldquo;<a href="http://ift.tt/2Azlg7g" class="local-link">This Week in Programming: Kotlin Eats Your Lunch</a>.&rdquo;</li>
<li>
<strong>GitHub Security Alerts:</strong> Further delivering on some promises it made last month to help developers keep track of and fix known flaws in dependencies, GitHub has <a href="http://ift.tt/2yMIDZJ" class="ext-link" rel="external nofollow">announced security alerts</a>, which it says will help with the &ldquo;over 75 percent of GitHub projects that have dependencies.&rdquo; The feature relies on one released at <a href="http://ift.tt/2nDIh2N" class="ext-link" rel="external nofollow">GitHub Universe</a> that helps <a href="http://ift.tt/2zPc6Uh" class="local-link">track dependencies</a>. With the dependency graph enabled, GitHub will notify users when it detects a vulnerability in one of their dependencies and offers suggested known fixes from the community. The new feature will also include vulnerabilities that have <a href="https://cve.mitre.org/" class="ext-link" rel="external nofollow">CVE IDs</a> (publicly disclosed vulnerabilities from the <a href="https://nvd.nist.gov/" class="ext-link" rel="external nofollow">National Vulnerability Database</a>).</li>
<li>
<strong>TensorFlow Lite:</strong> Google is also offering some follow-up on past promises with <a href="http://ift.tt/2AFdw2P" class="ext-link" rel="external nofollow">the developer preview of TensorFlow Lite</a>, which &ldquo;enables low-latency inference of on-device machine learning models.&rdquo; The &ldquo;lightweight solution for mobile and embedded devices&rdquo; works on both iOS and Android. TensorFlow Lite supports the <a href="http://ift.tt/2xnH83p" class="ext-link" rel="external nofollow">Android Neural Networks API</a> to take advantage of &ldquo;purpose-built custom hardware to process ML workloads more efficiently,&rdquo; but &ldquo;falls back to optimized CPU execution when accelerator hardware is not available, which ensures your models can still run fast on a large set of devices.&rdquo; Google says that &ldquo;going forward, TensorFlow Lite should be seen as the evolution of TensorFlow Mobile, and as it matures it will become the recommended solution for deploying models on mobile and embedded devices.&rdquo; The Google Research Blog also offered some peripheral insight, looking at <a href="http://ift.tt/2yBxPxv" class="ext-link" rel="external nofollow">on-device conversational modeling</a> with TensorFlow Lite. To help give developers a kick start, they have released an <a href="http://ift.tt/2igYMAg" class="ext-link" rel="external nofollow">on-device conversational model</a> and a <a href="http://ift.tt/2AFdZlB" class="ext-link" rel="external nofollow">demo app</a>.</li>
<li>
<strong>NumPy to Drop Python2:</strong> NumPy, the fundamental scientific computing package for Python, announced a <a href="http://ift.tt/2yDBZ7T" class="ext-link" rel="external nofollow">timeline for dropping Python 2 support</a>. Official support for Python 2 is <a href="http://ift.tt/29zkN8x" class="ext-link" rel="external nofollow">slated to end in 2020</a> and NumPy plans to follow a similar schedule, fully supporting Python2 and Python3 until December 31, 2018. Then, on January 1, 2019, any new feature releases will support only Python3, and long-term support (LTS) for Python2 will continue until December 31, 2019.</li>
<li>
<strong>Fortran Turns 60:</strong> First introduced in 1957, Fortran is celebrating its <a href="http://ift.tt/2hs69Eo" class="ext-link" rel="external nofollow">60th birthday</a>. Despite &ldquo;approaching retirement age,&rdquo; the language is going strong. The blog post explains that &ldquo;as the name implies, Fortran is designed to translate mathematical formulas into computer code. That explains its strong presence in fields that deal with a lot of mathematical formulas (particularly partial differential equations and the like).&rdquo;</li>
<li>
<strong>Go Turns 8:</strong> Meanwhile, it has been <a href="http://ift.tt/2zruZi9" class="ext-link" rel="external nofollow">8 years since Go was released</a> as an open source project and in that short time it has gathered nearly a million users worldwide and secured the #9 spot in GitHub&rsquo;s <a href="http://ift.tt/1huIPwK" class="ext-link" rel="external nofollow">2017 Octoverse</a> list of popular programming languages. Since its initial release, it has seen &ldquo;10 releases of the language, libraries and tooling with more than 1680 contributors making over 50,000 commits to the project&rsquo;s 34 repositories.&rdquo;</li>
</ul><h2>This Week in Language Trends</h2>
<ul>
<li>
<strong>Dramatic Technological Shifts:</strong> Stack Overflow is continuing with its series of data-driven blog posts examining the shifting terrain of technologies, this time trying to suss out the <a href="http://ift.tt/2iaCBvO" class="ext-link" rel="external nofollow">most dramatic shifts</a>. The two standouts for the last year include <a href="http://ift.tt/1nIq8hv" class="ext-link" rel="external nofollow">Swift</a>, Apple&rsquo;s language for iOS apps, and <a href="https://angular.io/" class="ext-link" rel="external nofollow">Angular</a>, Google&rsquo;s web framework. &ldquo;Both of these technologies grew incredibly fast to have a big impact,&rdquo; they reason, &ldquo;because they were natural next steps for existing developer communities.&rdquo;</li>
<li>
<strong>Java Devs Love Kotlin:</strong> The SDTimes reports that <a href="http://ift.tt/2ALDotY" class="ext-link" rel="external nofollow">Java developers are most excited to work with Kotlin</a>, citing a report by RebelLabs. According to the report, which surveyed 2,000 Java developers, Kotlin was &ldquo;the number one exciting technology for 2017 followed by Docker, Java 9, Spring, Angular, Java EE 8, Spring 5, Java 8, Kubernetes, and React.&rdquo;</li>
<li>
<strong>Scripting Language Popularity Slips:</strong> InfoWorld reports that <a href="http://ift.tt/2ijYp83" class="ext-link" rel="external nofollow">scripting languages are slipping in popularity</a> as developers &ldquo;opt for safer languages, despite ease of use offered by scripting languages like Perl, PHP, and Ruby.&rdquo; According to the <a href="http://ift.tt/2mbLj0F" class="ext-link" rel="external nofollow">TIOBE Index</a>, &ldquo;only Python is going strong&rdquo; and the cause of this decrease in popularity is likely run-time errors. &ldquo;Even a scripting language such as JavaScript that is inevitable while doing web programming was forced to evolve to a safer language,&rdquo; they write. &ldquo;Microsoft introduced a typed version of JavaScript called TypeScript and all kinds of frameworks such as Angular and React were developed to safeguard the language (and also add extra functionality).&rdquo;</li>
</ul><p class="attribution"><a href="http://bit.ly/2h30eoP" class="ext-link" rel="external nofollow">Microsoft</a> is a sponsor of The New Stack.</p>
<p class="attribution">Feature image via <a href="http://ift.tt/2hI1AcL" class="ext-image" rel="external nofollow">Wikimedia Commons</a>.</p>
<p>The post <a rel="nofollow" href="http://ift.tt/2zQozZO">This Week in Programming: Real-Time Collaborative Coding Comes to Atom and Visual Studio</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br><br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
November 18, 2017 at 09:00AM
